pam/authentication.go

```diff
@@ -32,9 +35,8 @@ func sendIsAuthenticated(ctx context.Context, client authd.PAMClient, sessionID,
 					access: responses.AuthCancelled,
 				}
 			}
-			return pamSystemError{
-				msg: fmt.Sprintf("Authentication status failure: %v", err),
-			}
+			return pam.NewTransactionError(pam.ErrSystem,
```

Comment:
I don’t think this is really idiomatic (and I see this part is not merged in upstream). This wrapper is doing an `errors.Join()` on both errors. You are joining separate errors, things that happens concurrently or sequentially, but the errors don’t have necessarily a relationship between each other).

Here, you are wrapping an error into another error. So you should use wrapping instead and then, you don’t need this constructor. If I look at your implemention in your go-pam fork (which is not merged upstream), then, you can and should directly get rid of the specific error type.

Basically, you just return an error wrapped into the other one, either as we did with the pamSystemError or with:
fmt.Errorf(`authentication status failure: %v (%w)`, err, pam.ErrSystem).

Then you can use `errors.Is` or `errors.As` to unwrap the underlying error.

I still feel the error system as it was initially implemented is more idiomatic.

Reply:
I've changed all this, `pam.TransactionError` is still eventually used as it allows better logging without doing things on our side. But it's not used elsewhere.